crates/router/src/routes/metrics/bg_metrics_collector.rs

```diff
@@ -0,0 +1,42 @@
+use storage_impl::redis::cache;
+
+const DEFAULT_BG_METRICS_COLLECTION_INTERVAL_IN_SECS: u64 = 15;
+
+pub fn spawn_metrics_collector(metrics_collection_interval_in_secs: &Option<u64>) {
+    let metrics_collection_interval = metrics_collection_interval_in_secs
+        .unwrap_or(DEFAULT_BG_METRICS_COLLECTION_INTERVAL_IN_SECS);
+
+    tokio::spawn(async move {
+        loop {
+            crate::gauge_metrics_for_imc!(
+                CONFIG_CACHE,
+                ACCOUNTS_CACHE,
+                ROUTING_CACHE,
+                CGRAPH_CACHE,
+                PM_FILTERS_CGRAPH_CACHE
+            );
+
+            tokio::time::sleep(std::time::Duration::from_secs(metrics_collection_interval)).await
+        }
+    });
+}
+
+#[macro_export]
+macro_rules! gauge_metrics_for_imc {
```

Comment:
You can avoid the `#[macro_export]` annotation, since it will only be used in this module. You would have to define the macro before its usage, since declarative macros are only accessible after they're defined.